command/agent/dns.go

```diff
@@ -628,7 +632,9 @@ func (d *DNSServer) handleRecurse(resp dns.ResponseWriter, req *dns.Msg) {
 	q := req.Question[0]
 	network := "udp"
 	defer func(s time.Time) {
-		d.logger.Printf("[DEBUG] dns: request for %v (%s) (%v)", q, network, time.Now().Sub(s))
+		d.logger.Printf("[DEBUG] dns: request for %v (%s) from client %s (%s)",
```

Comment:
Looks like you're expecting to interpolate 4 terms and are passing in 5.  I think the pattern for time is missing.